airflow/providers/docker/operators/docker.py

```diff
@@ -269,7 +269,12 @@ def _run_image(self) -> Optional[str]:
             # duplicated conditional logic because of expensive operation
             ret = None
             if self.do_xcom_push:
-                ret = self.cli.logs(container=self.container['Id']) if self.xcom_all else line.encode('utf-8')
+                if self.xcom_all:
+                    ret = self.cli.logs(container=self.container['Id'])
+                    if hasattr(ret, 'decode'):
+                        ret = ret.decode('utf-8')
+                else:
+                    ret = line.split('\n')[-1]
```

Comment:
Should we always do the decode? We are planning to drop pickle for XCom and default to using JSON as serialisation mechanism.

Reply:
this way it's consistent with the decoding before the logging (a couple of lines above) - but it also duplicates the decode call